Framework/PythonInterface/plugins/algorithms/HB3AAdjustSampleNorm.py

```diff
@@ -0,0 +1,272 @@
+                        issues['InputWorkspaces'] = "Workspace '{}' expected to have 3 dimensions".format(ws)
+
+        return issues
+
+    def PyExec(self):
+        load_van = not self.getProperty("VanadiumFile").isDefault
+        load_files = not self.getProperty("Filename").isDefault
+
+        has_van = not self.getProperty("VanadiumWorkspace").isDefault and not load_van
+
+        if load_files:
+            datafiles = self.getProperty("Filename").value
+        else:
+            datafiles = list(map(str.strip, self.getProperty("InputWorkspaces").value.split(",")))
+
+        vanadiumfile = self.getProperty("VanadiumFile").value
+        vanws = self.getProperty("VanadiumWorkspace").value
+        height = self.getProperty("DetectorHeightOffset").value
+        distance = self.getProperty("DetectorDistanceOffset").value
+        method = self.getProperty("OutputAsMDEventWorkspace").value
+
+        wavelength = -1
+        # Set the back-up wavelength value
+        wl_prop = self.getProperty("Wavelength")
+        if not wl_prop.isDefault:
+            wavelength = wl_prop.value
+
+        if method:
+            minvals = self.getProperty("MinValues").value
+            maxvals = self.getProperty("MaxValues").value
+            merge = self.getProperty("MergeInputs").value
+            wslist = []
+        else:
+            bin0 = self.getProperty("BinningDim0").value
+            bin1 = self.getProperty("BinningDim1").value
+            bin2 = self.getProperty("BinningDim2").value
+
+        out_ws = self.getPropertyValue("OutputWorkspace")
+        out_ws_name = out_ws
+
+        if load_van:
+            vanws = LoadMD(vanadiumfile, StoreInADS=False)
+
+        has_multiple = True if len(datafiles) > 1 else False
+
+        for file in datafiles:
+            if load_files:
+                scan = LoadMD(file, LoadHistory=False, StoreInADS=False)
+            else:
+                scan = mtd[file]
+
+            self.log().information("Processing file '{}'".format(file))
+
+            # If processing multiple files, append the base name to the given output name
+            if has_multiple:
+                if load_files:
+                    out_ws_name = out_ws + "_" + os.path.basename(file).strip(',.nxs')
+                else:
+                    out_ws_name = out_ws + "_" + file
+
+            exp_info = scan.getExperimentInfo(0)
+            self.__move_components(exp_info, height, distance)
+
+            # Get the wavelength from the file sample logs if it exists
+            if exp_info.run().hasProperty("wavelength"):
+                wavelength = exp_info.run().getProperty("wavelength").value
+            elif wavelength == -1:
+                raise RuntimeError("Wavelength not found in sample log and was not provided as input to algorithm")
+
+            # Use ConvertHFIRSCDtoQ (and normalize van), or use ConvertWANDSCtoQ which handles normalization itself
+            if method:
+                if has_van:
```

Comment:
If I provide a `VanadiumFile` then `has_van` is `False`  and therefore isn't applied.